test/integration/client-side-encryption/driver.test.ts

```diff
@@ -471,3 +482,353 @@ describe('Range Explicit Encryption with JS native types', function () {
+
+    beforeEach(async function () {
+      keyVaultClient = this.configuration.newClient();
+      await keyVaultClient.connect();
+      await keyVaultClient.db('keyvault').collection('datakeys');
+      const clientEncryption = new ClientEncryption(keyVaultClient, {
+        keyVaultNamespace: 'keyvault.datakeys',
+        kmsProviders: getKmsProviders()
+      });
+      dataKey = await clientEncryption.createDataKey('local');
+      setupClient = this.configuration.newClient();
+      await setupClient
+        .db()
+        .admin()
+        .command({
+          configureFailPoint: 'failCommand',
+          mode: 'alwaysOn',
+          data: {
+            failCommands: ['find'],
+            blockConnection: true,
+            blockTimeMS: 2000
+          }
+        } as FailPoint);
+    });
+
+    afterEach(async function () {
+      await keyVaultClient.close();
+      await setupClient
+        .db()
+        .admin()
+        .command({
+          configureFailPoint: 'failCommand',
+          mode: 'off'
+        } as FailPoint);
+      await setupClient.close();
+    });
+
+    const getKmsProviders = (): KMSProviders => {
+      const result = EJSON.parse(process.env.CSFLE_KMS_PROVIDERS || '{}') as unknown as {
+        local: unknown;
+      };
+
+      return { local: result.local };
+    };
+
+    const metadata: MongoDBMetadataUI = {
+      requires: {
+        mongodb: '>=4.2.0',
+        clientSideEncryption: '>=6.1.0'
+      }
+    };
+
+    context('when client is provided timeoutMS and command hangs', function () {
+      let encryptedClient: MongoClient;
+
+      beforeEach(async function () {
+        encryptedClient = this.configuration.newClient(
+          {},
+          {
+            autoEncryption: {
+              keyVaultClient,
+              keyVaultNamespace: 'keyvault.datakeys',
+              kmsProviders: getKmsProviders(),
+              schemaMap: {
+                'test.test': {
+                  bsonType: 'object',
+                  encryptMetadata: {
+                    keyId: [new UUID(dataKey)]
+                  },
+                  properties: {
+                    a: {
+                      encrypt: {
+                        bsonType: 'int',
+                        algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Random',
+                        keyId: [new UUID(dataKey)]
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            timeoutMS: 1000
+          }
+        );
+        await encryptedClient.connect();
+      });
+
+      afterEach(async function () {
+        await encryptedClient.close();
+      });
+
+      it('the command should fail due to a timeout error', metadata, async function () {
+        const err = await encryptedClient
+          .db('test')
+          .collection('test')
+          .insertOne({ a: 1 })
+          .catch(e => e);
+        expect(err).to.be.instanceOf(MongoOperationTimeoutError);
```

Comment:
We realized last week that it is important to measure the duration of the command when we expect it to time out.  If we expect it to time out in timeoutMS, then we need to confirm it actually took no more than timeoutMS to complete. 

There's some flexibility here, because CSOT isn't a hard gaurantee, but in other tests we've measured the duration (there's a measureDuration helper) and we assert its within a range of `timeoutMS`.  Can we do the same here?

Reply:
This applies to other places we're testing CSOT too

Reply:
added it in to all 5 csot enabled testing cases from this PR